pkg/acceptance/terrafarm/farmer.go

```diff
@@ -333,7 +338,15 @@ func (f *Farmer) Restart(ctx context.Context, i int) error {
 
 // Start starts the given process on the ith node.
 func (f *Farmer) Start(ctx context.Context, i int, process string) error {
-	_, _, err := f.execSupervisor(f.Nodes()[i], "start "+process)
+	name := "start " + process
+	_, _, err := f.execSupervisor(f.Nodes()[i], name)
+	if err == nil {
+		extendBy := f.NumNodes() - len(f.processes)
+		if extendBy > 0 {
```

Comment:
wouldn't all this new logic be simpler if you made `nodes` a `map[string][]string`?